### PR TITLE
Adding audit for theme-color in HTML

### PR DESCRIPTION
**Pull request by gauntface**
Might solve https://github.com/GoogleChrome/lighthouse/issues/25

cc @paulirish @paullewis

### CODE REVIEW

**Comment by paullewis** [comment]
This generally looks good. I think we should try and resolve #77 before we do any more HTML-based tests, but assuming you're happy to land that (and update this audit) we should be gtg!

File: test/audits/html/theme-colors.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                  <title>Sample page</title>
+                </head>
+                <body></body>
+              </html>`
+    }).value, false);
+  });
+
+  it('fails when multiple theme-colors exist', () => {
+    return assert.equal(Audit.audit({
+      html: `<!doctype html>
```
Comment (samccone) [inline test/audits/html/theme-colors.js]: can we DRY up these html mocks with a single function like

getMockHtml: function (insert) {
  return `...
   ...
  ${insert}
   ....
   ....`
}

**Comment by samccone** [comment]
nice work here @gauntface 1 comment and then I am :+1: to land

File: package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "chrome-devtools-frontend": "1.0.381789",
     "chrome-remote-interface": "^0.11.0",
     "devtools-timeline-model": "1.0.16",
+    "jsdom": "^8.1.0",
```
Comment (samccone) [inline package.json]: https://github.com/samccone/engine-deps

If we want to support older versions of node....

Comment (gauntface) [inline package.json]: Yeah, I couldn't even run lighthouse until I upgraded to 5.9 soooooooo

File: audits/mobile-friendly/viewport.js [hunk 3]
```diff
@@ -26,12 +26,19 @@ class Viewport {
...
-    // TODO: This'll be a false positive if the body content has this string.
-    // Unless we want to parse the original HTML, we should query for `head > .meta[name=viewport]`
-    const viewportExpression = /<meta ([^c]+)content="width=/gim;
+    let hasViewport = false;
+    if (inputs.window) {
+      const viewportElements =
+        inputs.window.document.querySelectorAll('head meta[name="viewport"]');
+
+      if (viewportElements.length === 1 &&
+        viewportElements[0].content.indexOf('width=') !== -1) {
```
Comment (samccone) [inline audits/mobile-friendly/viewport.js]: can we use getAttribute?

File: test/audits/html/theme-colors.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        <head>
+          ${headInsert}
+          <title>Sample page</title>
+        </head>
+        <body>
+          ${bodyInsert}
+        </body>
+      </html>`;
+
+    const jsdom = require('jsdom').jsdom;
```
Comment (samccone) [inline test/audits/html/theme-colors.js]: can we pull the require to the top of the file

Comment (gauntface) [inline test/audits/html/theme-colors.js]: will make that change.

Originally did this to try and please browserify but the ignore in gulp-browserify works regardless.

**Comment by paulirish** [comment]
We don't want to use jsdom as a dependency here. 

We will always have the browser available and its DOM is the source of truth. 

We have to be very conscious about minimizing the dependencies that we need at runtime. Pulling in jsdom and browserifying that into a chrome extension does not work for me.

**Comment by gauntface** [comment]
jsdom is ignored in the extension since it's not required, that's something.

What would you prefer to solve: https://github.com/GoogleChrome/lighthouse/issues/77

File: audits/mobile-friendly/viewport.js (hunk 3 above)
Comment (gauntface) [inline audits/mobile-friendly/viewport.js]: Just curious: Why getAttribute() over .content? I have no preference.

**Comment by samccone** [comment]
Looks like we need a rebase

**Comment by paulirish** [comment]
I read in the opensource handbook that you should upset everyone with a big rejection of a new contributor's first PR. 

Or not. :confused:    Again, sorry for being terse and stuff. 

I feed this thread more flowers for good luck: :tulip: :bouquet: :rose: 

-----------------

Discussion on the right approach will continue on #77.

**Comment by samccone** [comment]
going to close this for now, as it seems like there is a bit more exploration to do here.

@gauntface thanks so much for the PR, and this has totally helped us inform the direction we need to go when gathering and dealing with HTML!

I look forward to working more with you on this project!

**Comment by paulirish** [comment]
@gauntface I'm building off your PR over here: https://github.com/GoogleChrome/lighthouse/compare/master...paulirish:themecolor    Still a WIP I'll have something for you to review soon.

**Pull request closed without merging**